### ISSUE TITLE
Allow use of positional only syntax with Concatenate

### ISSUE THREAD

**Issue by cdce8p** [microsoft/pyright#2842]
**Is your feature request related to a problem? Please describe.**
PEP 612 mentions for the semantics of `Concatenate`:
```
The semantics of Concatenate[X, Y, P] are that it represents the parameters
represented by P with two positional-only parameters prepended.
```
https://www.python.org/dev/peps/pep-0612/#semantics

There has been a discussion in a [mypy PR](https://github.com/python/mypy/pull/11847) about requiring `/` in the wrapper signature to represent the `position-only` parameters. Doing so currently raises a type issue in pyright.

```py
from typing import Callable, Concatenate, ParamSpec

P = ParamSpec("P")


def decorator(
    func: Callable[P, None]
) -> Callable[Concatenate[int, P], None]:
    def wrapper(x: int, /, *args: P.args, **kwargs: P.kwargs) -> None:  # note the '/' between 'x' and '*args'
        ...
    return wrapper
```
```
error: Expression of type "(x: int, /, *args: P.args, **kwargs: P.kwargs) -> None" cannot be assigned to return type "(int, **P@decorator) -> None"
    Type "(x: int, /, *args: P.args, **kwargs: P.kwargs) -> None" cannot be assigned to type "(int, **P@decorator) -> None" (reportGeneralTypeIssues)
```

**Describe the solution you'd like**
By definition `x` is already a `positional-only` arg. Pyright shouldn't raise a type error when using `/` to archive proper runtime behavior.

**Additional context**
Without the `/` something like this would raise a runtime error.
```py
@decorator
def func(x: int) -> None:
    pass

reveal_type(func)  # (int, y: Any) -> None
func(1, x=2)
```
```
TypeError: decorator.<locals>.wrapper() got multiple values for argument 'x'
```

**Comment by erictraut** [comment]
Yeah, that makes sense.

This will be included in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.206, which I just published. It will also be in the next release of pylance.
